client/src/app/services/chart.service.ts

```diff
@@ -17,6 +17,26 @@ import {
   Tooltip
 } from "chart.js";
 
+// Type for indicator data returned from API
+interface IndicatorDataRow {
```

Comment:
Since window based resizing is really only affecting chart.js chart data types length, instead of caching and rebuilding full chart configurations again, instead of caching API data responses, we should cache the post processed chart dataset data in its native format and modify it only.  All charts and selections should be using an identical x-axis dataset.